### DISCUSSION TITLE
customFields relation - possible to define a relation to an entity defined in a custom plugin?

### DISCUSSION THREAD

**Discussion by devtrylah** [vendurehq/vendure#1501] [General]
In the spirit of leveraging the amazing Vendure engine as much as possible, we want to create a customField that defines a relation to an entity that is part of a custom plugin implementation.

We understand it is possible to define a relation to a Vendure entity, such as Asset (which has an example in the documentation). However, we've not had success trying to define one for entity from a plugin. 

The specific error we encountered was:

`[ExceptionHandler] The GraphQL type "XXX" specified by the Customer.YYY custom field does not exist`

But we know this GraphQL type XXX does exist. I suspect Vendure attempts to load this XXX type BEFORE loading the plugins, thus resulting in the error. But we can't confirm this indeed is the case. 

We would appreciate some advice or ideas on this please. Thank you in advance.

**Comment by michaelbromley** [comment]
Hi,

What you are attempting to do _should_ be possible, so it seems something is set up incorrectly. It's hard for me to say exactly what, based on the info provided so far, but you should check that the GraphQL type is made available to the APIs that are being extended by the custom field. This is a common error - let's say you define the type on your `adminApiExtensions` but then do not set `public: false` on the custom field definition. In that case, Vendure tries to extend the Shop API too but then errors because that GraphQL type is not defined on the Shop API.

**Comment by devtrylah** [comment]
Thanks for the hints! I was able to set up the relation. And I guess I will have to build a UI extension in order to populate the field (for instance into a dropdown list), as the relation is pointing to an entity defined in a custom plugin.

Is that correct? 

I copy and paste the relevant documentation extract below:

> UI for relation type
> 
> The Admin UI app has built-in selection components for “relation” custom fields which reference certain common entity types, such as Asset, Product, ProductVariant and Customer. If you are relating to an entity not covered by the built-in selection components, you will see a generic relation component which allows you to manually enter the ID of the entity you wish to select.
> 
> If the generic selector is not suitable, or is you wish to replace one of the built-in selector components, you can create a UI extension which defines a custom field control for that custom field. You can read more about this in the [custom form input guide](https://www.vendure.io/docs/plugins/extending-the-admin-ui/custom-form-inputs/)

**Reply by michaelbromley** [reply]
Yes, that's right. By default (in v1.5+) you should see the "generic" relation input, which allows you to directly set an ID. But ideally you'll want to implement a proper form control with better UX as described in those docs.
